server/routes/functionalSkillsResolver.ts

```diff
@@ -77,4 +77,8 @@ const assessmentsGroupedByTypeSortedByDateDesc = (assessments: Array<Assessment>
   return map
 }
 
-export { allFunctionalSkills, mostRecentFunctionalSkills }
+const functionalSkillsByType = (assessments: Array<Assessment>, type: Assessment['type']): Array<Assessment> => {
```

Comment:
That's cool 👍 , I didn't know you could reference a field to use it's type
```
type: Assessment['type']
```
(clearly the word "type" is overloaded in this comment, but I know what I'm trying to say 😁 . I'm trying to say its cool 😁 )

Reply:
I know we didnt talk about this earlier, but I wonder if this method should return the assessments sorted? Probably sorted by assessment date, given the UI use case is that the default sort order should be by date?

If we are going to sort them, then I think you'll want to destructure the array into a new array before filtering and sorting (I know we dont do that elsewhere in this class, but I think its possibly a bug, thinking about one of the problems we had with sorting arrays last week)
```
return [...assessments].filter(.....).sort(....)
```
And I think you can use the `dateComparator` that we already have 🤔

Reply:
Good early spot with the sort order, wouldn't have noticed that until the UI changes!